pkg/odo/cli/catalog/describe/component.go

```diff
@@ -0,0 +1,182 @@
+
+	if experimental.IsExperimentalModeEnabled() {
+		catalogDevfileList, err := catalog.ListDevfileComponents("")
+		if err != nil {
+			return err
+		}
+		for _, devfileComponent := range catalogDevfileList.Items {
+			if devfileComponent.Name == o.componentName {
+				o.devfileComponents = append(o.devfileComponents, devfileComponent)
+			}
+		}
+	}
+
+	return nil
+}
+
+// Validate validates the DescribeComponentOptions based on completed values
+func (o *DescribeComponentOptions) Validate() (err error) {
+	if len(o.devfileComponents) == 0 && o.component == "" {
+		return errors.Wrapf(err, "No components with the name \"%s\" found", o.componentName)
+	}
+
+	return nil
+}
+
+// Run contains the logic for the command associated with DescribeComponentOptions
+func (o *DescribeComponentOptions) Run() (err error) {
+	if experimental.IsExperimentalModeEnabled() {
+		w := tabwriter.NewWriter(os.Stdout, 5, 2, 3, ' ', tabwriter.TabIndent)
+		if log.IsJSON() {
+			if len(o.devfileComponents) > 0 {
+				for _, devfileComponent := range o.devfileComponents {
+					data, err := pkgUtil.DownloadFileInMemory(devfileComponent.Registry.URL + devfileComponent.Link)
+					if err != nil {
+						return errors.Wrapf(err, "Failed to download devfile.yaml for devfile component: %s", devfileComponent.Name)
+					}
+					devObj, err := devfile.ParseInMemory(data)
+					if err != nil {
+						return err
+					}
+
+					machineoutput.OutputSuccess(devObj)
+				}
+			}
+		} else {
+			if len(o.devfileComponents) > 1 {
+				log.Warningf("There are multiple components named \"%s\" in different multiple devfile registries.\n", o.componentName)
+			}
+			if len(o.devfileComponents) > 0 {
+				fmt.Fprintln(w, "Devfile Component(s):")
+
+				for _, devfileComponent := range o.devfileComponents {
+					fmt.Fprintln(w, "\n* Registry: "+devfileComponent.Registry.Name)
+
+					data, err := pkgUtil.DownloadFileInMemory(devfileComponent.Registry.URL + devfileComponent.Link)
+					if err != nil {
+						return errors.Wrapf(err, "Failed to download devfile.yaml for devfile component: %s", devfileComponent.Name)
+					}
+					devObj, err := devfile.ParseInMemory(data)
+					if err != nil {
+						return err
+					}
+					projects := devObj.Data.GetProjects()
+					// only print project info if there is at least one project in the devfile
+					if len(projects) > 0 {
+						fmt.Fprintln(w, "\nStarter Projects:")
+						for _, project := range projects {
+							yamlData, err := yaml.Marshal(project)
+							if err != nil {
+								return errors.Wrapf(err, "Failed to marshal devfile object into yaml")
+							}
+							fmt.Printf("---\n%s", string(yamlData))
+						}
+					} else {
+						fmt.Fprintln(w, "The Odo devfile component \""+o.componentName+"\" has no starter projects.")
+						yamlData, err := yaml.Marshal(devObj)
+						if err != nil {
+							return errors.Wrapf(err, "Failed to marshal devfile object into yaml")
+						}
+						fmt.Printf("---\n%s", string(yamlData))
+					}
```

Comment:
This also could be abstracted out in a function `printDevfileProjects()`, it would not overpopulate this function when we want to print more information.